homeassistant/components/enphase_envoy/__init__.py

```diff
@@ -24,6 +25,17 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     if not entry.unique_id:
         hass.config_entries.async_update_entry(entry, unique_id=envoy.serial_number)
 
+    if entry.unique_id != envoy.serial_number:
+        # If the serial number of the device does not match the unique_id
+        # of the config entry, it likely means the DHCP lease has expired
+        # and the device has been assigned a new IP address. We need to
+        # wait for the next discovery to find the device at its new address
+        # and update the config entry so we do not mix up devices.
+        raise ConfigEntryNotReady(
```

Comment:
Why don't we raise an issue for the user to give more instructions on what they can do to avoid this? (static IP, static DHCP lease)

Reply:
There are ~100 of these to do and most of the time the issue will fix itself within a few seconds as soon as statup finishes and discovery happens so its a bit of diminishing returns to add 100s of issue that will likely auto fix themselves

Reply:
Oh I didn't knew this happened using discovery, nice